test/lint/golangci.sh

```diff
@@ -1,6 +1,12 @@
 #!/bin/bash
 set -eu
 
+# Skip if already ran via golangci/golangci-lint-action
+if [ -n "${GITHUB_ACTIONS:-}" ]; then
+    echo "GitHub Action runner detected, skipping golangci-lint script (already done)"
+    exit 0
+fi
+
```

Comment:
If we arent using this on github any more we should remove the github references below.

Reply:
I'm happy with what it looks like now. It runs:

```
/home/runner/golangci-lint-2.1.6-linux-amd64/golangci-lint run  --timeout=5m --new --new-from-rev=74f662df66b345a3d2026df2d8b087436c1c13e4 --whole-files
```

Where 74f662df66b345a3d2026df2d8b087436c1c13e4 is the current main's HEAD which is the exact same commit the old script used to test with as can be seen in https://github.com/canonical/lxd/actions/runs/15452542118/job/43497869117#step:22:41